src/components/sidebar/routePathView/NewRoutePathView.tsx

```diff
@@ -4,68 +4,60 @@ import classnames from 'classnames';
 import { IRoutePath } from '~/models';
 import { Button } from '~/components/controls';
 import ButtonType from '~/enums/buttonType';
+import navigator from '~/routing/navigator';
 import { MapStore, NodeSize } from '~/stores/mapStore';
 import { NetworkStore } from '~/stores/networkStore';
 import { RoutePathStore } from '~/stores/routePathStore';
+import { RouteStore } from '~/stores/routeStore';
 import RoutePathFactory from '~/factories/routePathFactory';
 import ViewHeader from '../ViewHeader';
 import RoutePathViewForm from './RoutePathViewForm';
 import * as s from './routePathView.scss';
 
 interface INewRoutePathViewProps {
     mapStore?: MapStore;
+    routeStore?: RouteStore;
     routePathStore?: RoutePathStore;
     networkStore?: NetworkStore;
 }
 
-@inject('mapStore', 'routePathStore', 'networkStore')
+interface INewRoutePathViewState {
+    currentRoutePath?: IRoutePath;
+}
+
+@inject('mapStore', 'routeStore', 'routePathStore', 'networkStore')
 @observer
-class NewRoutePathView extends React.Component<INewRoutePathViewProps>{
-    initialRoutePath: IRoutePath;
+class NewRoutePathView extends React.Component<INewRoutePathViewProps, INewRoutePathViewState>{
+    currentRoutePath: IRoutePath;
 
     constructor(props: any) {
         super(props);
-        this.initialRoutePath = this.getInitialRoutePath();
-
-         // TODO: call this when newRoutePath button is clicked
-        const newRoutePath = RoutePathFactory.createNewRoutePath('1', '1');
-        this.props.routePathStore!.setRoutePath(newRoutePath);
-    }
 
-    // TODO: Use routePath store instead of state
-    // TODO, this is just a placeholder, implement real logic for creating new routePaths
-    private getInitialRoutePath = () => {
-        const newRoutePath: IRoutePath = {
-            internalId: '',
-            routeId: '',
-            routePathName: 'Uusi reitinsuunta',
-            routePathNameSw: 'Ny ruttriktning',
-            direction: '1',
-            positions: [[0, 0]],
-            geoJson: null,
-            visible: true,
-            startTime: new Date,
-            endTime: new Date,
-            lastModified: new Date,
-            routePathLinks: [],
-            originFi: '',
-            originSw: '',
-            destinationFi: '',
-            destinationSw: '',
-            routePathShortName: '',
-            routePathShortNameSw: '',
-            modifiedBy: '',
+        this.state = {
+            currentRoutePath: undefined,
         };
-        return newRoutePath;
     }
 
-    // TODO
-    public onChange = () => {
+    public onChange = (property: string) => (value: string) => {
+        this.setState({
+            currentRoutePath: { ...this.state.currentRoutePath!, [property]: value },
+        });
     }
 
-    // TODO
     public onSave = () => {
+        // TODO
+    }
+
+    componentWillMount() {
+        if (!this.props.routePathStore!.routePath) {
```

Comment:
Usually we use function names to describe what a section does. This becomes a problem when we use lifecycle methods, since they don't describe what the functionality does. I would suggest extracting the code in the lifecycle method in to something like initEmptyRoutePath() or similar

Reply:
Or just adding a comment here could do the trick

Reply:
A wise man once said: Comments should be unnecessary

Reply:
Already fixed this